Formula/m/mumps.rb

```diff
@@ -0,0 +1,56 @@
+class Mumps < Formula
+  desc "MUltifrontal Massively Parallel sparse direct Solver"
+  homepage "https://mumps-solver.org/"
+  url "https://mumps-solver.org/MUMPS_5.6.2.tar.gz"
+  sha256 "13a2c1aff2bd1aa92fe84b7b35d88f43434019963ca09ef7e8c90821a8f1d59a"
+  license "CECILL-C"
+
+  # Core dependencies
+  depends_on "gcc"
+  depends_on "openblas"
+  fails_with :clang
+
+  def install
+    makefile = "Makefile.G95.SEQ"
+    cp "Make.inc/" + makefile, "Makefile.inc"
+    inreplace "Makefile.inc", "-soname", "-install_name" unless OS.linux?
+    inreplace "Makefile.inc", ".so", ".dylib" unless OS.linux?
+    inreplace "Makefile.inc", "# RPATH_OPT = -Wl,-rpath,/path/to/MUMPS_x.y.z/lib/", "RPATH_OPT = -Wl,-rpath,#{lib}"
+    make_args = ["CDEFS=-DAdd_", "OPTF=-fallow-argument-mismatch"]
+    make_args += ["CC=#{ENV.cc} -fPIC",
+                  "FC=gfortran -fPIC -fopenmp",
+                  "FL=gfortran -fPIC -fopenmp"]
+    # Default lib args
+    # Note: MUMPS link cannot find LAPACK without these
+    # lines
+    blas_lib = "-L#{Formula["openblas"].opt_lib} -lopenblas"
+    make_args << "LIBBLAS=#{blas_lib}"
+    make_args << "LAPACK=#{blas_lib}"
```

Comment:
Can you add a comment with the upstream issue report for this? Because it shouldn't need this much custom configuration to compile right?

Reply:
I find it pretty common that makefile systems rely on non-trivial user intervention to correctly setup the make variables for a given build environment. MUMPS seems to take that approach.
The `OPTF` and `CDEFS` args are actually specifically required by the MUMPS build instructions docs, as are the blas and lapack specs and most of the extra compile flags. From the MUMPS build docs: _"in most cases, Makefile.inc should be adapted to fit with your architecture, libraries and compilers"_ . I absolutely agree there are improvements than can/should be made to the MUMPS build system, and I have reached out to the MUMPS devs.

I have reached out to the MUMPS devs about an install target, and a MacOS specific makefile, as well as a few of the places I do think we should not have to specify a make argument. _However_ the MUMPS developers do not maintain as public facing issue tracker, VCS, or archive of any kind AFAICT (from googling and some communication on the general mailing list), so there unfortunately is nothing for you folks to monitor, track, or reference. I am communicating with them through their dev mailing list, and general mailing list, which is also how I will submit patches if and when there are any.  I understand you would like to avoid the added maintenance of arguments specified to a build system, so how would you like to proceed here given this context?
To be clear, I am happy to upstream or iterate on whatever is definitively necessary to upstream to make this PR work for homebrew, but with no tracking, I think perhaps this PR can move in tandem with my upstreams to MUMPS.
Maybe there's someone at homebrew I can CC on some of the MUMPS threads?

Additionally, all that said, the lack of `-lblas` in `Make.inc/Makefile.G95.SEQ` is almost certainly a bug, and I will include that in my iterations with the MUMPS devs.

Reply:
> I find it pretty common that makefile systems rely on non-trivial user intervention to correctly setup the make variables for a given build environment

No, it's really not that common if you look at the software inside of Homebrew.